src/MvvmNano.Core/IPresenter.cs

```diff
@@ -10,6 +10,12 @@ namespace MvvmNano
     public interface IPresenter
     {
         /// <summary>
+        /// Changes the root page to the view of the given View model.
+        /// </summary>
+        /// <typeparam name="TViewModel"></typeparam>
+        void ChangeRootViewModel<TViewModel>() where TViewModel : MvvmNanoViewModel;
```

Comment:
Ah, I see where you are coming from: https://github.com/aspnetde/MvvmNano/pull/14/files#diff-78efc21b1935c7f63c842e460a51af1aR56

But: While you _can_ do this (technically), you should simply call `NavigateTo<MasterViewModel>()` here, which is provided by the base class.

And that's what you basically want to do: Telling where to go (Master), not how to get there (replacing the root view). That _how_ is something that belongs to the presenter.

Some background on my motivation here: By leaving this inside of the presenter, it's possible to keep view models clean and mix up UI even with native elements (view controllers or activities). Master detail is even a good example here: on Android you may want to display both, master and detail, next to each other, when you're on a tablet, but on two different activities, when you're on a handheld device.